vendor/processors/stripe.py

```diff
@@ -1,104 +1,261 @@
-            # print('Type is: %s' % e.error.type)
-            # print('Code is: %s' % e.error.code)
-            # # param is '' in this case
-            # print('Param is: %s' % e.error.param)
-            # print('Message is: %s' % e.error.message)
+                    # Each product needs an attached price obj. Check if one exists for the product
+                    # and attach it to the mapping or create one and attach
+                    existing_price_status, price_response = self.check_price_does_exist(product_id)
+                    if existing_price_status and product_response:
+                        new_price_status, price_id = self.get_price_id_with_product(product_id)
+                    else:
+                        new_price_status, price_id = self.create_price_with_product(product_id)
 
+                    self.products_mapping[product_name] = {
+                        'product_id': product_id,
+                        'price_id': price_id
+                    }
+
+    def stripe_call(self, func, **func_args):
+        try:
+            return func(**func_args)
+        except stripe.error.CardError as e:
+            # Since it's a decline, stripe.error.CardError will be caught
+            logger.error(e.user_message)
+            return None
         except stripe.error.RateLimitError as e:
             # Too many requests made to the API too quickly
-            # pass
-            payment.result = '{"message":"Rate Limit Error"}'
-
+            logger.error(e.user_message)
+            return None
         except stripe.error.InvalidRequestError as e:
             # Invalid parameters were supplied to Stripe's API
-            payment.result = '{"message":"Invalid Parameters"}'
-
+            logger.error(e.user_message)
+            return None
         except stripe.error.AuthenticationError as e:
             # Authentication with Stripe's API failed
             # (maybe you changed API keys recently)
-            payment.result = '{"message":"Not Authenticated"}'
-
+            logger.error(e.user_message)
+            return None
         except stripe.error.APIConnectionError as e:
             # Network communication with Stripe failed
-            payment.result = '{"message":"Network Error"}'
-
+            logger.error(e.user_message)
+            return None
         except stripe.error.StripeError as e:
             # Display a very generic error to the user, and maybe send
             # yourself an email
-            payment.result = '{"message":"Something Went Wrong.  You were not charged, please try again."}'
-
+            logger.error(e.user_message)
+            return None
         except Exception as e:
             # Something else happened, completely unrelated to Stripe
             # TODO: Send email to self
-            payment.result = '{"message":"A serious error has occured.  Our team has been notified."}'
+            logger.error(e.user_message)
+            return None
```

Comment:
I would remove the comments that are before the logger lines. They seem redundant as the Exception Name is clear enough